ehrql/utils/sqlalchemy_exec_utils.py

```diff
@@ -75,6 +75,12 @@ def execute_with_retry(*args, **kwargs):
                     raise
                 retries += 1
                 log(f"{e.__class__.__name__}: {e}")
+                # Note that as we're running with AUTOCOMMIT isolation level, this
+                # rollback should be purely internal to SQLAlchemy and should have no
+                # effect on the database itself. See:
+                # https://docs.sqlalchemy.org/en/20/core/connections.html#understanding-the-dbapi-level-autocommit-isolation-level
```

Comment:
Good grief.  This both pragmatic and horrid.

Reply:
IKR? :disappointed: